### PR TITLE
Enable PerformanceTests #882

### PR DESCRIPTION
**Pull request by basilevs**
Enable PerformanceTests for Pull Requests labeled "performance".
See [discussion](https://github.com/eclipse-platform/eclipse.platform.swt/pull/903#discussion_r1405455957) on why tests can't be enabled on a permanent basis.

### CODE REVIEW

**Comment by laeubi** [comment]
@basilevs just in case you are interested, I could think about triggering a workflow by adding a label to a PR that then additionally executes the performance tests on demand.

**Comment by laeubi** [comment]
Yes one need to develop such workflow on the a fork as it requires to merge things to master (can be reset later on of course) and have additional permissions, but as you seem  very enthusiastic on this maybe you like to go in that direction as well :+1:

**Comment by laeubi** [comment]
Even though I approved the Run it seems the Github actions failed to execute, do you like to take a look?

**Comment by basilevs** [comment]
> Even though I approved the Run it seems the Github actions failed to execute, do you like to take a look?

I can't find logs or errors.

**Comment by basilevs** [comment]
I've missed a colon in step declaration. Would be nice to get an error from that.

**Comment by laeubi** [comment]
> I've missed a colon in step declaration. Would be nice to get an error from that.

yes that a bit unfortunate but you should receive an email if there is a fatal error in the action, also you can edit the file with github UI and then gets a validation.

**Comment by basilevs** [comment]
> Yes one need to develop such workflow on the a fork as it requires to merge things to master (can be reset later on of course) and have additional permissions, but as you seem very enthusiastic on this maybe you like to go in that direction as well 👍

@laeubi could you please mark [my pull requests](https://github.com/eclipse-platform/eclipse.platform.swt/pulls/basilevs) with [performance](https://github.com/eclipse-platform/eclipse.platform.swt/pulls?q=is%3Aopen+is%3Apr+author%3Abasilevs+label%3Aperformance) label?

**Comment by laeubi** [comment]
@basilevs have you tested this with your fork? I think even if I aprove the workflow run your changes are not picked up maybe because you are not a cmoitter on the repo.

**Comment by basilevs** [comment]
API Tools fail the build. What is the workflow here? [Last time I've tried to bump versions](https://github.com/eclipse-platform/eclipse.platform.swt/pull/883), it did not end well.

**Comment by iloveeclipse** [comment]
```
[ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.4:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.swt: Only qualifier changed for (org.eclipse.swt/3.124.200.v20231127-1131). Expected to have bigger x.y.z than what is available in baseline (3.124.200.v20231113-1355)
```

This means, this is first change in the release in this bundle but manifest version is not updated.
Depending on the change you have to bump either last version segment (if the API os not changed) or second one (if API is affected).

**Comment by basilevs** [comment]
> ```
> [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.4:compare-version-with-baselines (compare-attached-artifacts-with-release) on project org.eclipse.swt: Only qualifier changed for (org.eclipse.swt/3.124.200.v20231127-1131). Expected to have bigger x.y.z than what is available in baseline (3.124.200.v20231113-1355)
> ```
> 
> This means, this is first change in the release in this bundle but manifest version is not updated. Depending on the change you have to bump either last version segment (if the API os not changed) or second one (if API is affected).

As before no code was changed at all (the change modifies only .github directory).
I've created https://github.com/eclipse-platform/eclipse.platform.swt/pull/909 to bump the version

**Comment by akurtakov** [comment]
@basilevs 3 years ago perf tests have been migrated from storing both (current and latest release) timings in a db (this database server was a nightmare to maintain and historical data with OS updates and etc. proved to be useless) to using the test files which contain timings to show the diffs in perfermance . This also degraded over time:
* 4.27 (March 2023) - working for a few perf tests  https://download.eclipse.org/eclipse/downloads/drops4/R-4.27-202303020300/performance/performance.php . SWT tests worked (to some extend) https://download.eclipse.org/eclipse/downloads/drops4/R-4.27-202303020300/performance/basicPerformance.php?name=org.eclipse.swt&build=I20230302-0300&baseline=R-4.26-202211231800
* 4.28 (June 2023) - only baseline results https://download.eclipse.org/eclipse/downloads/drops4/R-4.28-202306050440/performance/performance.php
* 4.29 (September 2023) - no perf tests data at all

Long story short perf tests were running (minus infra/releng failures) till March this year . The way these perf tests worked was to run the same test twice - once with previous release and once with latest I-build and compare timings. The disabled assertions were actually failing based on "timing" criteria from DB content which as said previously is not that useful with fast changing environments nowadays.

**Comment by basilevs** [comment]
Skipped the slow test on MacOS with `Assume`.

**Comment by basilevs** [comment]
@laeubi this works fine. Please merge.
